harness/determined/cli/experiment.py

```diff
@@ -593,7 +593,7 @@ def set_gc_policy(args: Namespace) -> None:
                 c["uuid"],
                 render.format_resources(c["resources"]),
             ]
-            for c in sorted(checkpoints, key=lambda c: (c["trial_id"], c["step_id"]))
+            for c in sorted(checkpoints, key=lambda c: (c["trial_id"], c["step"]["id"]))
```

Comment:
i think we just shouldn't be using step id anymore (checkpoints soon won't be tied to a step, so this will change completely)
```suggestion
            for c in sorted(checkpoints, key=lambda c: (c["trial_id"], c["end_time"]))
```